src/Windows/Core/Impl/Shell/ICoreShell.cs

```diff
@@ -12,7 +11,12 @@ namespace Microsoft.Common.Core.Shell {
     /// composition container, export provider, global VS IDE
     /// services and so on.
     /// </summary>
-    public interface ICoreShell : ICompositionCatalog {
+    public interface ICoreShell : IIdleTimeService, ICompositionCatalog {
+        /// <summary>
+        /// Application-global services access
+        /// </summary>
+        IServiceContainer GlobalServices { get; }
```

Comment:
I am thinking about making *single* ICoreShell with `Services` and extensions to get `Log`, `Process`, etc via `GetService<T>`. Similarly, `IEditorShell` that is mapping command targets turns into 'controller service`. `IAppConstants` split into Name/Locale which go to ICoreShell and the rest goes to `IPlatformService`. Then we don't need `IApplicationShell`.